app/src/lib/stores/app-store.ts

```diff
@@ -1694,20 +1690,13 @@ export class AppStore extends TypedBaseStore<IAppState> {
     )
     this.showSideBySideDiff = getShowSideBySideDiff()
 
-    this.automaticallySwitchTheme = getAutoSwitchPersistedTheme()
-
-    if (this.automaticallySwitchTheme) {
-      this.selectedTheme = isDarkModeEnabled()
-        ? ApplicationTheme.Dark
-        : ApplicationTheme.Light
-      setPersistedTheme(this.selectedTheme)
-    } else {
-      this.selectedTheme = getPersistedTheme()
-    }
+    this.selectedTheme = getPersistedTheme()
+    this.automaticallySwitchTheme =
+      this.selectedTheme === ApplicationTheme.System
+    setPersistedTheme(this.selectedTheme)
```

Comment:
Setting the persisted theme right after reading it, feels a bit odd to me 🤔 

After looking at the implementation I understood why, but maybe it should be documented with a comment.